app/components/search-box.ts

```diff
@@ -19,6 +19,15 @@ function generateSpaceToken(): Token {
   return new Token('', ' ', null);
 }
 
+let passiveEventSupport = false;
+try {
+  let options = { get passive() { return passiveEventSupport = true; } };
+  window.addEventListener('test', options as any, options);
+  window.removeEventListener('test', options as any, options as any);
+} catch(err) {
+  passiveEventSupport = false;
+}
```

Comment:
Do you mind explaining this a little?

Reply:
Yeah, I should add a comment here. The short version is that:
1️⃣ The browser can optimize performance of a mouse wheel observer if it knows you'll never cancel it -- thus, the `passive` option.
2️⃣ Chrome complains loudly if you don't have the passive option set
3️⃣ IE doesn't support passing options to `addEventListener` 😡 
4️⃣ So ... we have to create a dummy object that will indicate whether we can pass an object to `addEventListener`; the computed property flips the switch if it's accessed. This allows us to be passive if we're not running in IE.

The code was lifted almost entirely from MDN, so I'll put a link to the relevant part of the documentation there.